kontonummer.js

```diff
@@ -404,24 +409,38 @@
             return {
                 is_valid: false,
                 errors: errors,
-                matched_banks: matched_banks
+                matched_banks: matched_banks,
+                has_warnings: Boolean(warnings.length)
             };
         }
 
         return {
             is_valid: is_valid,
             errors: [],
-            matched_banks: matched_banks
+            matched_banks: matched_banks,
+            has_warnings: Boolean(warnings.length)
         };
     };
 
     var validateLength = function(bank, bankNumber) {
         var errors = [];
 
         if (bank.regex.test(bankNumber)) {
-            if (bankNumber.length < bank.lengths.clearing + bank.lengths.account) {
+            let actualBankNumber = bankNumber;
+            let minBankNumberLength =  bank.lengths.clearing + bank.lengths.account;
+            const maxBankNumberLength = bank.lengths.clearing + bank.lengths.account;
+            
+            if(bank.lengths.min_account) {
+                // Since account number is zero padded we need to slice it before trim
+                const onlyAccountNumber = actualBankNumber.slice(bank.lengths.clearing);
+                const onlyClearingNumber = actualBankNumber.substr(0, bank.lengths.clearing);
+                actualBankNumber = `${onlyClearingNumber}${parseInt(onlyAccountNumber, 10)}`;
+                minBankNumberLength = bank.lengths.clearing + bank.lengths.min_account; 
```

Comment:
`bank.lengths.min_account` only refers to the account number, not the clearing number.

Linting - avoid trailing whitespace:

```suggestion
                minBankNumberLength = bank.lengths.min_account;
```